### PR TITLE
Open twitter tapping on tweet

### LINKED ISSUE #129: Open Twitter on tapping tweet

### PR DESCRIPTION
**Pull request by maxxx777**
The PR closes #129 issue.
`selectMessageTrigger` was added to `MessagesTableViewController` to handle select row in table view.
`LinkResponsiveTextView` was introduced to let open links as by selecting a table view cell as selecting a link in table view cell.
No UI changes.

### Screenshots (before the changes)
IMG_0007
IMG_0008

### Screenshots (after the changes)
IMG_0010
IMG_0009

### Before merging the PR

- [x] If applicable, did you add a before / after screenshot of the feature?
- [x] Did you update the [CHANGELOG.md](https://github.com/criticalmaps/criticalmaps-ios/blob/master/CHANGELOG.md)?

### CODE REVIEW

File: CriticalMass/TwitterViewController.swift [hunk 1]
```diff
@@ -50,3 +53,19 @@ class TwitterViewController: UIViewController {
         messagesTableViewController.tableView.separatorInset = UIEdgeInsets(top: 0.0, left: 73.0, bottom: 0.0, right: 0.0)
     }
 }
+
+private extension TwitterViewController {
+    func openTweet(_ tweet: Tweet) {
+        if let webURL = tweet.webURL, UIApplication.shared.canOpenURL(webURL) {
+            UIApplication.shared.open(webURL, options: [:], completionHandler: nil)
```
Comment (mltbnz) [inline CriticalMass/TwitterViewController.swift]: Could we also add the universal link variant of the user has installed the Twitter app? Not entirely sure if that even works with the tweet link but would be great. Then we could route to the app if `canOpenUrl` is true and if not we open the Link in Safari or if it was for me could also be SafariViewController. 

This is also fine but maybe you could check if twitter app link is possible

Comment (mltbnz) [reply CriticalMass/TwitterViewController.swift]: I rechecked on my phone and it already works like a charm 🦢
Let's merge (once the conflict is resolved)

Comment (maxxx777) [reply CriticalMass/TwitterViewController.swift]: the web link should open the Twitter app if It's installed. So no need to add the twitter universal link.

**Comment by maxxx777** [comment]
@lennet @mltbnz thanks for the comments. will fix conflicts for this and other PRs later this month when be back after vacations :)

**Comment by maxxx777** [comment]
- Fixed the conflicts 
- Added highlighting when cell selected

Simulator Screen Shot - iPhone 8 Plus - 2019-10-17 at 15 17 25
Simulator Screen Shot - iPhone 8 Plus - 2019-10-17 at 15 17 48

File: CriticalMass/TweetTableViewCell.swift [hunk 2]
```diff
@@ -68,6 +68,12 @@ class TweetTableViewCell: UITableViewCell, MessageConfigurable, IBConstructable
         handleLabel.text = "@\(tweet.user.screen_name)"
         userNameLabel.text = tweet.user.name
         userImageView.sd_setImage(with: URL(string: tweet.user.profile_image_url_https), placeholderImage: UIImage(named: "Avatar"))
+        if #available(iOS 13.0, *) {
+            // workaround for iOS 13
+            // set background color to clear to keep the same color for background text view and background cell view
+            // when the cell is highlighted
```
Comment (lennet) [inline CriticalMass/TweetTableViewCell.swift]: is the textview backgroundcolor on pre iOS 13 clear by default?

Comment (maxxx777) [reply CriticalMass/TweetTableViewCell.swift]: I set `Default` color in storyboard. It looks like it `white` in pre iOS 13 (`UIExtendedGrayColorSpace 1 1`). 
But if I set `white`,  it stays `white` in iOS 13 when cell selected:

Simulator Screen Shot - iPhone 8 Plus - 2019-10-17 at 16 08 22

**Pull request merged**